### ISSUE TITLE
 context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function

### ISSUE THREAD

**Issue by chinmay241** [internet-sicherheit/caliper-docker#6]
```
2020.04.30-10:18:03.719 info  [caliper] [client.js] 	Existing 1 clients will be reused in next test round... 
2020.04.30-10:18:03.722 info  [caliper] [local-client.js] 	txUpdateTime: 1000
2020.04.30-10:18:04.769 info  [caliper] [local-client.js] 	Info: client 70409 init test :transfering money
2020.04.30-10:18:05.772 info  [caliper] [local-client.js] 	Info: client 70409 start test runFixedNumber():transfering money
2020.04.30-10:18:05.773 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 1
2020.04.30-10:18:05.774 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:05.794 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 2
2020.04.30-10:18:05.795 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:05.813 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 3
2020.04.30-10:18:05.813 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:05.834 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 4
2020.04.30-10:18:05.835 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:05.854 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 5
2020.04.30-10:18:05.855 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:05.875 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 6
2020.04.30-10:18:05.875 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:05.895 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 7
2020.04.30-10:18:05.895 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:05.914 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 8
2020.04.30-10:18:05.915 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:05.937 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 9
2020.04.30-10:18:05.938 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:05.953 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 10
2020.04.30-10:18:05.954 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:05.975 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 11
2020.04.30-10:18:05.976 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:05.994 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 12
2020.04.30-10:18:05.995 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.014 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 13
2020.04.30-10:18:06.015 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.034 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 14
2020.04.30-10:18:06.035 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.054 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 15
2020.04.30-10:18:06.056 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.073 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 16
2020.04.30-10:18:06.074 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.093 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 17
2020.04.30-10:18:06.094 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.113 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 18
2020.04.30-10:18:06.114 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.133 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 19
2020.04.30-10:18:06.133 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.154 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 20
2020.04.30-10:18:06.155 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.174 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 21
2020.04.30-10:18:06.175 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.195 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 22
2020.04.30-10:18:06.197 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.213 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 23
2020.04.30-10:18:06.214 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.235 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 24
2020.04.30-10:18:06.235 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.254 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 25
2020.04.30-10:18:06.254 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.275 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 26
2020.04.30-10:18:06.275 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.293 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 27
2020.04.30-10:18:06.294 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.313 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 28
2020.04.30-10:18:06.314 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.333 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 29
2020.04.30-10:18:06.334 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.354 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 30
2020.04.30-10:18:06.355 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.373 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 31
2020.04.30-10:18:06.373 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.395 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 32
2020.04.30-10:18:06.396 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.414 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 33
2020.04.30-10:18:06.414 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.435 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 34
2020.04.30-10:18:06.435 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.454 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 35
2020.04.30-10:18:06.455 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.474 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 36
2020.04.30-10:18:06.475 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.494 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 37
2020.04.30-10:18:06.494 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.514 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 38
2020.04.30-10:18:06.514 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.534 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 39
2020.04.30-10:18:06.534 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.554 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 40
2020.04.30-10:18:06.555 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.573 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 41
2020.04.30-10:18:06.574 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.594 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 42
2020.04.30-10:18:06.595 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.614 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 43
2020.04.30-10:18:06.615 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.635 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 44
2020.04.30-10:18:06.636 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.658 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 45
2020.04.30-10:18:06.659 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.674 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 46
2020.04.30-10:18:06.675 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.694 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 47
2020.04.30-10:18:06.695 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.714 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 48
2020.04.30-10:18:06.715 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.734 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 49
2020.04.30-10:18:06.735 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.753 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 50
2020.04.30-10:18:06.753 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.774 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 51
2020.04.30-10:18:06.775 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.795 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 52
2020.04.30-10:18:06.795 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.813 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 53
2020.04.30-10:18:06.815 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.834 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 54
2020.04.30-10:18:06.835 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.855 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 55
2020.04.30-10:18:06.855 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.874 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 56
2020.04.30-10:18:06.875 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.893 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 57
2020.04.30-10:18:06.894 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.914 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 58
2020.04.30-10:18:06.914 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.934 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 59
2020.04.30-10:18:06.934 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.953 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 60
2020.04.30-10:18:06.954 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.974 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 61
2020.04.30-10:18:06.974 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:06.994 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 62
2020.04.30-10:18:06.994 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.014 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 63
2020.04.30-10:18:07.015 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.034 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 64
2020.04.30-10:18:07.034 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.055 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 65
2020.04.30-10:18:07.055 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.074 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 66
2020.04.30-10:18:07.074 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.095 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 67
2020.04.30-10:18:07.095 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.115 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 68
2020.04.30-10:18:07.115 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.134 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 69
2020.04.30-10:18:07.135 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.154 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 70
2020.04.30-10:18:07.154 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.175 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 71
2020.04.30-10:18:07.175 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.194 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 72
2020.04.30-10:18:07.195 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.214 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 73
2020.04.30-10:18:07.215 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.234 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 74
2020.04.30-10:18:07.235 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.255 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 75
2020.04.30-10:18:07.256 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.273 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 76
2020.04.30-10:18:07.274 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.293 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 77
2020.04.30-10:18:07.294 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.313 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 78
2020.04.30-10:18:07.313 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.334 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 79
2020.04.30-10:18:07.335 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.355 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 80
2020.04.30-10:18:07.356 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.373 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 81
2020.04.30-10:18:07.373 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.395 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 82
2020.04.30-10:18:07.396 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.415 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 83
2020.04.30-10:18:07.415 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.435 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 84
2020.04.30-10:18:07.436 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.455 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 85
2020.04.30-10:18:07.455 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.475 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 86
2020.04.30-10:18:07.475 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.495 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 87
2020.04.30-10:18:07.496 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.515 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 88
2020.04.30-10:18:07.516 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.534 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 89
2020.04.30-10:18:07.534 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.553 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 90
2020.04.30-10:18:07.554 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.573 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 91
2020.04.30-10:18:07.574 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.594 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 92
2020.04.30-10:18:07.595 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.614 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 93
2020.04.30-10:18:07.614 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.634 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 94
2020.04.30-10:18:07.634 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.653 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 95
2020.04.30-10:18:07.654 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.673 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 96
2020.04.30-10:18:07.673 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.695 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 97
2020.04.30-10:18:07.695 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.714 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 98
2020.04.30-10:18:07.715 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.734 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 99
2020.04.30-10:18:07.735 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.754 error [caliper] [ethereum.js] 	Failed tx on simple calling method transfer nonce 100
2020.04.30-10:18:07.755 error [caliper] [ethereum.js] 	TypeError: context.contracts[contractID].methods[methodCall.verb].estimateGas is not a function
2020.04.30-10:18:07.977 info  [caliper] [report-builder] 	### Test result ###
2020.04.30-10:18:07.978 info  [caliper] [report-builder] 	
+----------+------+------+-----------------+-----------------+-----------------+-----------------+------------------+
| Name     | Succ | Fail | Send Rate (TPS) | Max Latency (s) | Min Latency (s) | Avg Latency (s) | Throughput (TPS) |
|----------|------|------|-----------------|-----------------|-----------------|-----------------|------------------|
| transfer | 0    | 100  | 50.5            | 0.00            | 100000.00       | NaN             | 0.0              |
+----------+------+------+-----------------+-----------------+-----------------+-----------------+------------------+


```

**Comment by kiview** [comment]
So this is happening when executing the benchmarks in our currently only scenario? If I look at [this](https://ethereum.stackexchange.com/questions/21654/correct-syntax-for-estimating-gas-cost-of-a-function-in-web3) SO answer, I assume we have some mismatch in web3j versions here.

**Comment by kiview** [comment]
So, API from docs:
```javascript
myContract.methods.myMethod([param1[, param2[, ...]]]).estimateGas(options[, callback])
```

How caliper-ethereum uses the API:
```javascript
context.contracts[contractID].methods[methodCall.verb](...methodCall.args).estimateGas();
```

This, potentially, does not fit. But it might also related to #1.

**Comment by kiview** [comment]
This is fixed implicitly by #1 and #9.

**Comment by kiview** [comment]
Closed by #9.
